### DISCUSSION TITLE
I want to in WPF Trrigger property Ischeck is true  i set background color but In avalonia is Pseudoclasses .How to Customer? like follow picture

### DISCUSSION THREAD

**Discussion by hubing53** [AvaloniaUI/Avalonia#8962] [Q&A] [answered]
![微信截图_20220914220142](https://user-images.githubusercontent.com/112156001/190175875-a163bb14-ab05-438c-bbef-6c212f4a68cc.png)

**Comment by jp2masa** [comment, marked as answer]
Did you try `Selector="local|NavButtonImage:checked"`?

**Reply by timunie** [reply]
Well, what is NavButton based on? Is it a ToggleButton? If yes, your selector is similar to this: 

https://github.com/AvaloniaUI/Avalonia/blob/62a5ee14fb12b26672ea25a06b2154b19fd83a01/src/Avalonia.Themes.Fluent/Controls/ToggleButton.xaml#L66-L88

if it's your own control and you define your own pseudo-class remember to add a `:` in front of the pseudo class name. 

Happy coding
Tim

**Reply by hubing53** [reply]
Finally ICustomer Pseudo-class I Use follow Syntax
![1663334044713_610ddea2](https://user-images.githubusercontent.com/112156001/190647230-bef6855d-2311-47bc-9251-91549e2f506e.png)
my XamlCode like this:
 <Style Selector="local|NavButtonImage:checked">
        <Setter Property="Background">
            <Setter.Value>
                <LinearGradientBrush StartPoint="0,0" EndPoint="1,0">
                    <GradientStop Offset="0" Color="#20D6BE"></GradientStop>
                    <GradientStop Offset="0.55" Color="#1ECD98"></GradientStop>
                </LinearGradientBrush>
            </Setter.Value>
        </Setter>
        <Setter Property="GeoFillColor" Value="White"></Setter>
    </Style>
    
    <Style Selector="local|NavButtonImage:unchecked:mouseenter">
        <Setter Property="Background" Value="#DDDDDD"></Setter>
    </Style>
   
    <Style Selector="local|NavButtonImage:unchecked:mouseleave">
        <Setter Property="Background" Value="#F6F6F6"></Setter>
    </Style>
My C# Code Like this:
  #region PseudoClassCustomer
        protected override void OnPointerEnter(PointerEventArgs e)
        {
            base.OnPointerEnter(e);
            PseudoClasses.Set(":mouseenter", true);
            PseudoClasses.Set(name: ":mouseleave", false);
        }

        protected override void OnPointerLeave(PointerEventArgs e)
        {
            base.OnPointerLeave(e);
            PseudoClasses.Set(":mouseenter", false);
            PseudoClasses.Set(name: ":mouseleave", true);
        }

        protected override void OnChecked(RoutedEventArgs e)
        {
            base.OnChecked(e);
            PseudoClasses.Set(":checked", true);
            PseudoClasses.Set(name: ":unchecked", false);
        }

        protected override void OnUnchecked(RoutedEventArgs e)
        {
            base.OnUnchecked(e);
            PseudoClasses.Set(":checked", false);
            PseudoClasses.Set(name:":unchecked", true);
        }
        #endregion
finally the effect result like this:
![1663334252242_4bae848a](https://user-images.githubusercontent.com/112156001/190647916-85ecdee4-9975-4f5a-bf92-cee635329ac6.png)
